mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/delivery/InVmEventDelivery.java

```diff
@@ -79,7 +79,7 @@ public InVmEventDelivery(MetricFactory metricFactory) {
 
     private Mono<Void> doDeliverToListener(MailboxListener.ReactiveMailboxListener mailboxListener, Event event) {
         if (mailboxListener.isHandling(event)) {
-            return Mono.defer(() -> Mono.from(metricFactory.runPublishingTimerMetric(timerName(mailboxListener),
+            return Mono.defer(() -> Mono.from(metricFactory.decoratePublisherWithTimerMetric(timerName(mailboxListener),
```

Comment:
This defer is evil and should not exist. See https://github.com/linagora/james-project/pull/3434/files?file-filters%5B%5D=.java&file-filters%5B%5D=.scala#r438615250

Reply:
ok, it should not relate to my PR.I will take care of it in another PR with this https://github.com/linagora/james-project/pull/3434/files?file-filters%5B%5D=.java&file-filters%5B%5D=.scala#r438615250

Reply:
Yes please <3

Reply:
Where is the JIRA ticket?

Reply:
Fixed here, as we write tests demonstrating it affects this pull request